src/subcommands/crates.rs

```diff
@@ -27,22 +47,68 @@ pub fn crates(args: Vec<String>, max_age: std::time::Duration) -> Result<(), std
         publishers.sort_unstable_by_key(|p| (p.kind, p.login.clone()));
     }
 
-    println!("\nDependency crates with the people and teams that can publish them to crates.io:\n");
-    for (i, (crate_name, publishers)) in ordered_owners.iter().enumerate() {
-        let pretty_publishers: Vec<String> = publishers
-            .iter()
-            .map(|p| match p.kind {
-                PublisherKind::team => format!("team \"{}\"", p.login),
-                PublisherKind::user => p.login.to_string(),
-            })
-            .collect();
-        let publishers_list = comma_separated_list(&pretty_publishers);
-        println!("{}. {}: {}", i + 1, crate_name, publishers_list);
-    }
+    if output_json {
+        let mut jdoc = CratesJdoc{
+            warning: concat! (
+                "Note: there may be outstanding publisher invitations. crates.io provides no way to list them. ",
+                "See https://github.com/rust-lang/crates.io/issues/2868 for more info."
+            ).to_string(),
+            description: concat! ("Dependency crates with the people and teams that ",
+                "can publish them to crates.io"
+            ).to_string(),
+            crates: HashMap::new()
+        };
+        for (i, (crate_name, publishers)) in ordered_owners.iter().enumerate() {
+            let user_publishers: Vec<String> = publishers
+                .iter()
+                .map(|p| match p.kind {
+                    PublisherKind::team => "".to_string(),
+                    PublisherKind::user => p.login.to_string(),
+                })
+                .filter(|x| !x.is_empty())
+                .collect();
+            let team_publishers: Vec<String> = publishers
+                .iter()
+                .map(|p| match p.kind {
+                    PublisherKind::team => p.login.to_string(),
+                    PublisherKind::user => "".to_string(),
+                })
+                .filter(|x| !x.is_empty())
+                .collect();
+            if jdoc.crates.contains_key(&crate_name.to_string()) {
+                panic!("Collision detected in crates.io names for crate {}. Results unreliable.")
```

Comment:
This produces a compiler warning because there's nothing to substitute `{}` with in the `panic!` macro.